### DISCUSSION TITLE
Could theia support cpptools fully

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-theia/theia#13864] [Q&A]
Hi,
    I have ever used cdt-gdb-adapter to debug my gdb c++ application, but cdt-gdb-adapter has not been updated for seven months,  and nobody answers questions on its discussion. when it comes to complex debug situations, current  cdt-gdb-adapter version 0.108 can not meet requirements. 
    so I try to use cpptools vs code extension in my theia application. but some feature not support in theia when install cpptools extension
    1. can not see memory.bin of variable in theia but ok in vscode
![image](https://github.com/eclipse-theia/theia/assets/4537809/d19b5818-aac3-442c-846b-8161d1b407a3)
   2. can not debug in disassembly view but ok in vscode
![image](https://github.com/eclipse-theia/theia/assets/4537809/9595b8e3-8a6e-43e9-9cfd-b018ca2d86be)
could theia support cpptools fully in future or could I do something to make it support

**Comment by msujew** [comment]
Hey @wss29,

1. Currently, I don't think that Theia supports debugging in the disassembly/memory view. Since we don't have a feature proposal for this, do you mind creating an issue for this?
2. While we could support cpptools in theory (after all, it's just using the vscode API), we currently don't have plans to support it, because the [cpptools distribution license](https://marketplace.visualstudio.com/items/ms-vscode.cpptools/license) forbids installing it outside of VSCode due to some of the proprietary binaries included in there. Installing the vsix file makes you liable for violating the license agreement.

**Reply by planger** [reply]
Just as an additional pointer:
There is an [upcoming PR](https://github.com/eclipse-cdt-cloud/vscode-memory-inspector/pull/139) for the Memory Inspector (VSX version) that supports data breakpoints.

@wss29 Which unanswered questions in the discussions do you mean? I only saw [your feature request](https://github.com/eclipse-cdt-cloud/cdt-gdb-adapter/issues/320) on Chinese characters in the paths of a launch config. Is that the one you are referring to?

For this in particular, I think it is currently not planned, but we'd be thrilled if you consider contributing this feature directly or with the [professional support options in Theia](https://theia-ide.org/support/) or [CDT Cloud](https://eclipse.dev/cdt-cloud/support/).

**Reply by wss29** [reply]
Hi @planger 
    What confuses me now is the 'cwd' config does not work in cdt-gdb-adapter.
    1. I have a cpp-debug-workspace project in D disk
    2. I run task 'compile example' to build a.exe. set breakpoint in a.cpp and debug it, everything is ok
    3. I copy D:\\cpp-debug-workspace  to D:\\cpp-debug-workspace1
    4. I set "cwd": "D:\\cpp-debug-workspace1" in launch.json
    5. debug it again, I get error "No source file named d:\\cpp-debug-workspace1\\a.cpp." but the file is indeed there
![image](https://github.com/eclipse-theia/theia/assets/4537809/cf81a909-96f9-41a2-93c4-9261ebd41c1a)
    6. I have seen https://github.com/eclipse-cdt-cloud/cdt-gdb-adapter/pull/306,  it says 'cwd' work, but I tried it, and not work 
    7. I use [cdd-gdb-](https://open-vsx.org/extension/eclipse-cdt/cdt-gdb-vscode) version 0.108 in windows 10 system to debug
[cpp-debug-workspace.zip](https://github.com/user-attachments/files/16052740/cpp-debug-workspace.zip)

**Reply by wss29** [reply]
Hi @msujew 
   I have created an issue https://github.com/eclipse-theia/theia/issues/13865  
  in fact use cpptools in theia is not a good choice, so hope to get more support from cdt-gdb

**Reply by planger** [reply]
@wss29 I suggest to amend this finding to [your issue](https://github.com/eclipse-cdt-cloud/cdt-gdb-adapter/issues/320). The error message could be related to the CWD of the GDB process, as well as with the symbol table that is created during the debug build.

**Reply by wss29** [reply]
Hi @planger 
    I create a new issue at  [329](https://github.com/eclipse-cdt-cloud/cdt-gdb-adapter/issues/329), [320](https://github.com/eclipse-cdt-cloud/cdt-gdb-adapter/issues/320) is another issue
